lib/Command/ConfigReport.php

```diff
@@ -30,8 +30,25 @@
  */
 class ConfigReport extends Command {
 
-	public function __construct() {
+	/**
+	 * @var ReportDataCollector
+	 */
+	private $reportDataCollector;
 
+	public function __construct() {
```

Comment:
Can't you inject `ReportDataCollector` here or is that the place where DI failed ?
You can try registering an alias in Application:
```php
$container->registerAlias('OCA\ConfigReport\ReportDataCollector', 'ReportDataCollector');
```

So when the DI code sees that the constructor needs "OCA\ConfigReport\ReportDataCollector" it will query that. But since you registered it as "ReportDataCollector" (without namespace) it won't find it. The alias would help.